src/lib.rs

```diff
@@ -111,6 +111,37 @@ impl<T> NonEmpty<T> {
     pub fn iter<'a>(&'a self) -> impl Iterator<Item = &T> + 'a {
         std::iter::once(&self.0).chain(self.1.iter())
     }
+
+    /// Often we have a `Vec` but want to ensure that it is `NonEmpty` before
+    /// proceeding with a computation. Using `from_vec` will give us a proof
+    /// that we have a `NonEmpty` in the `Some` branch, otherwise it allows
+    /// the caller to handle the `None` case.
+    ///
+    /// # Example Use
+    ///
+    /// ```
+    /// use nonempty::NonEmpty;
+    ///
+    /// let non_empty_vec = NonEmpty::from_vec(vec!([1, 2, 3, 4, 5]));
+    /// assert!(non_empty_vec.is_some());
+    ///
+    /// let empty_vec: Option<NonEmpty<u32>> = NonEmpty::from_vec(Vec::new());
+    /// assert!(empty_vec.is_none());
+    /// ```
+    pub fn from_vec(vec: Vec<T>) -> Option<NonEmpty<T>> {
+        let mut vec = vec;
+        let head = vec.pop();
```

Comment:
This will pop the *last* element of the vec, I don't think that's what you want?